src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/CosmosDBSourceTask.java

```diff
@@ -90,8 +91,17 @@ public void start(Map<String, String> map) {
                 if(node == null) {
                     continue;
                 }
+
+                // Set the Kafka message key if option is provided and "id" field is set in document
+                String messageKey = "";
```

Comment:
id should always be set in this case as the document comes from Cosmos DB and it's not possible to not have an id field in Cosmos DB. 

if you do add in the ability to choose the xpath for messageKeyPath then this check should remain. 

if messageKeyPath does not exist in the document, what should we do? fail? ignore?

Reply:
will do, and if it doesn't exist, i think leaving the key as blank would be a good fallback